### PR TITLE
Fix issue #29, use bytes instead of nibbles

### PR DESCRIPTION
**Pull request by ScottPJones**
This adds a fast getindex as well.

### CODE REVIEW

**Comment by ScottPJones** [comment]
This fixes problems brought up in issue #29

**Review by xiaodaigh — APPROVED** [review]

File: src/base.jl [hunk 1]
```diff
@@ -134,6 +143,14 @@ for T in (UInt1024, UInt512, UInt256, UInt128, UInt64, UInt32)
     end
 end
 
+# These are simply for backwards compatibility reasons
+#const ss30_str = ss31_str
+#const ss62_str = ss63_str
+#const ss126_str = ss127_str
+const ShortString30  = ShortString31
+const ShortString62  = ShortString63
+const ShortString126 = ShortString127
```
[the remark this answers is not in the document]
Comment (ScottPJones) [reply src/base.jl:152]: Done!

File: src/base.jl [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+using Base: @_inline_meta, @propagate_inbounds, @_propagate_inbounds_meta
 
 struct ShortString{T} <: AbstractString where {T}
     size_content::T
 end
 
 # check if a string of size `sz` can be stored in ShortString{T}`
 function check_size(T, sz)
-    max_len = sizeof(T) - size_nibbles(T)  # the last few nibbles are is used to store the length
+    max_len = sizeof(T) - size_bytes(T)  # the last few nibbles are is used to store the length
```
[the remark this answers is not in the document]
Comment (ScottPJones) [reply src/base.jl:14]: Done!

File: src/base.jl [hunk 3]
```diff
@@ -134,6 +143,14 @@ for T in (UInt1024, UInt512, UInt256, UInt128, UInt64, UInt32)
     end
 end
 
+# These are simply for backwards compatibility reasons
+#const ss30_str = ss31_str
+#const ss62_str = ss63_str
+#const ss126_str = ss127_str
```
[the remark this answers is not in the document]
Comment (ScottPJones) [reply src/base.jl:149]: Thanks for your input!

**Pull request merged**